[merge 3] Add a new Customizing The Deployment procedure (#73)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/con_manifest-overrides.adoc

// Module included in the following assemblies:
//
// <List assemblies here, each on a new line>

// This module can be included from assemblies using the following include statement:
// include::<path>/con_manifest-overrides.adoc[leveloffset=+1]

// The file name and the ID are based on the module title. For example:
// * file name: con_my-concept-module-a.adoc
// * ID: [id='con_my-concept-module-a_{context}']
// * Title: = My concept module A
//
// The ID is used as an anchor for linking to the module. Avoid changing
// it after the module has been published to ensure existing links are not
// broken.
//
// The `context` attribute enables module reuse. Every module's ID includes
// {context}, which ensures that the module has a unique ID even if it is
// reused multiple times in a guide.
//
// In the title, include nouns that are used in the body text. This helps
// readers and search engines find information quickly.
// Do not start the title with a verb. See also _Wording of headings_
// in _The IBM Style Guide_.
[id="manifest-overrides_{context}"]
= Customizing The Deployment

> The Service Telemetry Operator watches for a `ServiceTelemetry` manifest to be loaded into {OpenShiftShort} which then creates other objects in memory to result in the dependent Operators creating the workloads they are responsible for managing.

Reviewer: To clarify, when the ServiceTelemetry manifest loads into OKD, does the Service Telemetry Operator then create other objects. When these objects are created, dependent Operators create their own workloads. 

I think it's clear that I don't understand and perhaps there's another way to write this sentence.

Author: @JoanneOFlynn2018 that's exactly what happens, you got it :)

Reviewer: Oh! I was sure I wasn't interrupting it correctly. Phew and hooray!

WARNING: When overriding the manifest the entire manifest contents including object names or namespaces must be provided. No dynamic parameter substitution will be done when overriding manifests.

> The easiest way to override manifests in a way that will work with {ProjectShort} is to first deploy a default environment using the core options only (see <<creating-a-servicetelemetry-object-in-openshift>>). After {ProjectShort} is deployed, use `oc get` to retrieve the default deployed manifest. This will help keep manifests compatible with the other manifests managed by the Service Telemetry Operator.

Reviewer: ```suggestion
To override manifests in a way that works with {ProjectShort}, deploy a default environment using the core options only. For more information about the core options, see <<creating-a-servicetelemetry-object-in-openshift>>. When you deploy {ProjectShort}, use the `oc get` command to retrieve the default deployed manifest. This helps to keep manifests compatible with the other manifests that are managed by the Service Telemetry Operator.
```

> To override manifests in a way that works with {ProjectShort}, deploy a default environment using the core options only. For more information about the core options, see <<creating-a-servicetelemetry-object-in-openshift>>. When you deploy {ProjectShort}, use the `oc get` command to retrieve the default deployed manifest. This helps to keep manifests compatible with the other manifests that are managed by the Service Telemetry Operator.

Author: ```suggestion
To override manifests in a way that works with {ProjectShort}, deploy a default environment using the core options only. For more information about the core options, see <<creating-a-servicetelemetry-object-in-openshift>>. When deploying {ProjectShort}, use the `oc get` command to retrieve the default deployed manifest. Using a manifest originally generated by Service Telemetry Operator helps to keep manifests compatible with the other objects that are managed by the Operators.
```

Author: @JoanneOFlynn2018 additional tweaking

For example, when the `metricsEnabled: true` parameter is configured in the `ServiceTelemetry` manifest, the Service Telemetry Operator will request components for metrics retrieval and storage using the default manifests. In some cases however you may want to override the default manifest. The available manifest override parameters available are:

|===

> | Override Parameter | Description | Retrieval command

Reviewer: This table is great. Very clear and concise info
